nefertari/scripts/es.py

```diff
@@ -153,107 +149,80 @@ def recreate_index(registry):
-        self._pool = {process_id: [] for process_id in range(0, processes)}
-        self.options = options
-
-        for model_name in model_names:
-            for process_id in range(0, processes):
-                self._pool[process_id].append(ESTask(model_name, options))
+def split_collection(limit, n, collection):
+    list_size = (math.ceil(limit / n) - 1) or 1
+    iteration = 0
 
-    def get_task(self, process_id):
-        if not self.is_empty(process_id):
-            return self._pool[process_id].pop()
-        return None
+    for i in range(0, limit, list_size):
+        iteration += 1
 
+        if iteration == n:
+            yield collection[i:]
+            break
 
-    def is_empty(self, process_id):
-        return len(self._pool[process_id]) == 0
+        yield collection[i: i + list_size]
 
 
-class ESProcessor(object):
-
-    def __init__(self, options, pool, process_id, model_names, processes_count):
-        self.options = options
-        self.pool = pool
-        self.id = process_id
-        self.processes_count = processes_count
-        self.model_names = model_names
+def get_processing_queue(model_names, processes_count, options):
+    manager = Manager()
+    queue = manager.Queue()
 
-    @staticmethod
-    def split_collection(limit, n, collection):
-        list_size = (math.ceil(limit / n) - 1) or 1
-        iteration = 0
-
-        for i in range(0, limit, list_size):
-            iteration += 1
-
-            if iteration == n:
-                yield collection[i:]
-                break
-
-            yield collection[i: i + list_size]
-
-    def get_chunks(self):
-        model_chunks = {}
+    def _get_chunks(model_names, processes_count, q):
+        setup_app(options, put_mappings=False)
         from nefertari import engine
         from pyramid_sqlalchemy import Session
+        model_chunks = {}
 
-        for model_name in self.model_names:
+        for model_name in model_names:
             model = engine.get_document_cls(model_name)
             limit = model.get_collection().count()
             table_name = model.__tablename__
             statement = text('SELECT {} FROM public.{}'.format(model.pk_field(), table_name))
             query = Session().query(model).from_statement(statement)
-            items = list(sorted(query.values(model.pk_field())))
-            chunks = list(self.split_collection(limit, self.processes_count, items))
+            items = list(map(lambda item: getattr(item, model.pk_field()) ,sorted(query.values(model.pk_field()))))
+            chunks = list(split_collection(limit, processes_count, items))
+            model_chunks[model_name] = chunks
 
-            if len(chunks) - 1 < self.id:
-                model_chunks[model_name] = tuple()
-                continue
+            for p in range(0, processes_count):
+                if len(chunks) > p:
+                    q.put((model_name, chunks[p]), block=False)
```

Comment:
Do you do something with `model_chunks` afterwards? I don't quite understand what this method does